cmd/weaver-server/main.go

```diff
@@ -58,7 +57,7 @@ func startWeaver(_ *cli.Context) error {
 		log.Printf("StartServer: failed to initialise etcd route loader: %s", err)
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), (1 * time.Second))
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
This is a hack to fix the other problem

Reply:
I see there are two ways. 
one is not to set `context.withTimeout` in the main. Because this is being passed to watcher in https://github.com/gojektech/weaver/blob/2884d7103a398ed3491153ff356e65db5307c173/server/server.go#L31
and as per watch doc

```go
type Watcher interface {
	// Next blocks until an etcd event occurs, then returns a Response
	// representing that event. The behavior of Next depends on the
	// WatcherOptions used to construct the Watcher. Next is designed to
	// be called repeatedly, each time blocking until a subsequent event
	// is available.
	//
	// If the provided context is cancelled, Next will return a non-nil
	// error. Any other failures encountered while waiting for the next
	// event (connection issues, deserialization failures, etc) will
	// also result in a non-nil error.
	Next(context.Context) (*Response, error)
}
```
so this will returns non nil error which causes router to stop watching updates.
https://github.com/gojektech/weaver/blob/2884d7103a398ed3491153ff356e65db5307c173/etcd/routeloader.go#L89-L93


2. Other approach is to construct separate `context` in server.go and pass it router to watch for updates. IMHO this will be bad because any information from main context will not be passed to router. 

I can't think of any other approaches at this point. Do you have any suggestions?